include/ada/idna/to_unicode.h

```diff
@@ -0,0 +1,12 @@
+
+#ifndef ADA_IDNA_TO_UNICODE_H
+#define ADA_IDNA_TO_UNICODE_H
+
+#include <string>
+#include <string_view>
+
+namespace ada::idna {
+std::string to_unicode(const std::string_view& input);
+}  // namespace ada::idna
+
+#endif  // ADA_IDNA_TO_UNICODE_H
```

Comment:
It will complain here that the file does end with an empty line. :-)

Reply:
> @miguelteixeiraa Can you check the CI errors?
> 
> I think your PR looks good. It might even be correct (minus the build errors). I am not 100% sure what to_unicode is supposed to do. Maybe you have checked the specification? I haven't.

Yes, I've checked:
https://www.rfc-editor.org/rfc/inline-errata/rfc3490.html
and also https://www.rfc-editor.org/rfc/inline-errata/rfc3492.html

Based on: https://www.rfc-editor.org/rfc/inline-errata/rfc3490.html
The step 2 would be
2. Perform the steps specified in [NAMEPREP] and fail if there is an
      error.  (If step 3 of ToASCII is also performed here, it will not
      affect the overall behavior of ToUnicode, but it is not
      necessary.)  The AllowUnassigned flag is used in [NAMEPREP].
      
So the NAMEPREP thing is missing (which involves things like normalization, and checks for the label validity (sizes...etc etc )) 

I'm doing 1, 3, 4, 5, 6, 7 (through the tests), 8

   1. If all code points in the sequence are in the ASCII range (0..7F)
      then skip to step 3.

   2. Perform the steps specified in [NAMEPREP] and fail if there is an
      error.  (If step 3 of ToASCII is also performed here, it will not
      affect the overall behavior of ToUnicode, but it is not
      necessary.)  The AllowUnassigned flag is used in [NAMEPREP].

   3. Verify that the sequence begins with the ACE prefix, and save a
      copy of the sequence.

   4. Remove the ACE prefix.

   5. Decode the sequence using the decoding algorithm in [PUNYCODE] and
      fail if there is an error.  Save a copy of the result of this
      step.

   6. Apply ToASCII.

   7. Verify that the result of step 6 matches the saved copy from step
      3, using a case-insensitive ASCII comparison.

   8. Return the saved copy from step 5.
   
  This is the algo to be applied to each label. It is basically the reverse operation that is being done for to_ascii.

Reply:
We should have a clang-format rule for this :)